tiledb/sm/group/group_directory.cc

```diff
@@ -128,22 +125,42 @@ Status GroupDirectory::load() {
   // Some processing is also done here for things that don't depend on others.
   // List (in parallel) the root directory URIs
   tasks.emplace_back(tp_->execute([&]() {
-    auto&& [st, uris] = list_root_dir_uris();
-    RETURN_NOT_OK(st);
-
-    root_dir_uris = std::move(uris.value());
-
-    return Status::Ok();
+    try {
```

Comment:
We don't need `Status` return to use the thread pool. A `void` lambda function is sufficient.

Reply:
I'm getting `Error C2440 'return': cannot convert from 'std::future<void>' to 'std::future<tiledb::common::Status>' (compiling source file C:\Users\teo\code\TileDB\TileDB\sm\group\group_directory.cc) TILEDB_CORE_OBJECTS C:\Users\teo\code\TileDB\tiledb\common\thread_pool\thread_pool.h 115`.

Reply:
I added a `return Status::Ok()` for now and let the exception flow.